articles/documenting-node-js-rest-api-using-swagger/index.md

```diff
@@ -0,0 +1,577 @@
+
+To document this path, we have to edit the `/docs/todos/delete-todo.js` file as follows:
+
+```js
+module.exports = {
+  // operation's method.
+  delete: {
+    tags: ["Todo CRUD operations"], // operation's tag
+    description: "Deleting a todo", // short desc
+    operationId: "deleteTodo", // unique operation id
+    parameters: [
+      // expected parameters
+      {
+        name: "id", // name of param
+        in: "path", // location of param
+        schema: {
+          $ref: "#/components/schemas/id", // id model
+        },
+        required: true, // mandatory
+        description: "Deleting a done todo", // param desc
+      },
+    ],
+    // expected responses
+    responses: {
+      // response code
+      200: {
+        description: "Todo deleted successfully", // response desc
+      },
+      // response code
+      404: {
+        description: "Todo not found", // response desc
+      },
+      // response code
+      500: {
+        description: "Server error", // response desc
+      },
+    },
+  },
+};
+```
+
+From above, we have:
+
+- Specified the method of operation, `delete`.
+- Specified the `tags` of the operation.
+- Described the operation with a short `description` and `operationId`.
+- Specified the `parameters`. For a parameter, we have described the `name`, `in`, `schema`, `required`, and `description` properties.
+- Specified the `responses`. In this operation, we can get a `200` response if the todo is deleted successfully, a `404` response if the todo with that `id` is not found, and a `500` response if there is a server error deleting the todo.
+
+To test this:
+
+- Ensure that the development server is running.
+- In your browser, refresh the documentation page.
+- In case of errors, revisit the steps. Else, the following section should be added to your page:
+
+![delete_todo_screenshot](/engineering-education/documenting-node.js-rest-api-using-swagger/delete-todo-screenshot.png)
+
+- To try this out, get the `id` of the updated todo, click the `Try it out` button, paste the `id` in the parameters section, click the `Execute` button, and then observe the server response.
+
+### Summary
+
+In this article, we have documented a simple Node.js REST API using swagger by following the below steps:
+
+- [Documenting API's general information](#documenting-api's-general-information)
+- [Documenting API's servers](#documenting-api's-servers)
+- [Documenting API's tags](#documenting-api's-tags)
+- [Documenting API's components](#documenting-api's-components)
+- [Documenting API's paths](#documenting-api's-paths)
+
+### Conclusion
+
+Documenting an API serves vast advantages towards the usability of the API since anyone can understand and consume it. Swagger does the heavy lifting process in documenting a restful API. Depending on the API you are building, swagger offers vast [functionalities](https://swagger.io/docs/specification/about/).
```

Comment:
```suggestion
Documenting an API serves vast advantages towards the usability of the API since anyone can understand and consume it. Swagger does the heavy lifting process in documenting a RESTful API. Depending on the API you are building, swagger offers vast [functionalities](https://swagger.io/docs/specification/about/).
```

Reply:
fixed.